### PR TITLE
Trigyn testcases

### PR DESCRIPTION
**Pull request by gitdivyanshu**

### CODE REVIEW

File: app/javascript/components/action-button/component.spec.js [hunk 1]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
...
+    mountedComponent(<ActionButton {...newProps} />, state);
+    expect(screen.getAllByRole('button', { className: 'isTransparent' })).toHaveLength(1);
+   
+  });
 
   // Check if outlined class is set
-  it.todo("renders outlined button");
+  //it.todo("renders outlined button");
+
+  //it.todo("renders a pending state");
```
Comment (jtoliver-quoin) [inline app/javascript/components/action-button/component.spec.js:52]: Uncomment this line, so devs will know where test need to be added

File: app/javascript/components/action-button/component.spec.js [hunk 2]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
...
+      isTransparent: true
+    }; 
+    
+    mountedComponent(<ActionButton {...newProps} />, state);
+    expect(screen.getAllByRole('button', { className: 'isTransparent' })).toHaveLength(1);
+   
+  });
 
   // Check if outlined class is set
-  it.todo("renders outlined button");
```
Comment (jtoliver-quoin) [inline app/javascript/components/action-button/component.spec.js:41]: Remove this line

**Review by jtoliver-quoin — COMMENTED** [review]
There are some other component test in the action-button directory that need to be added to this pr.

./components/action-button/components/default-button/component.unit.test.js
./components/action-button/components/link/component.unit.test.js
./components/action-button/components/icon-button/component.unit.test.js

File: app/javascript/components/action-button/component.spec.js [hunk 3]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
...
+  it("should contain .cancel class", () => {
+    const newProps = {
+      ...props,
+      cancel:true
+    };
 
-  it.todo("triggers a passed cancel function");
+   mountedComponent(<ActionButton {...newProps} />, state);
+   expect(screen.getAllByRole('button', { className: 'cancel' })).toHaveLength(1);
+
```
[the remark this answers is not in the document]
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js]: done

File: app/javascript/components/action-button/component.spec.js (hunk 1 above)
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js]: I have uncommented these line and move these line at the end .

File: app/javascript/components/action-button/component.spec.js (hunk 2 above)
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js:41]: done.

File: app/javascript/components/action-button/component.spec.js [hunk 4]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
...
+  it("renders an icon button", () => {
+    const newProps = {
+      ...props,
+      className: "MuiSvgIcon-root"
+    };
+
+    mountedComponent(<ActionButton {...newProps} />, state);
+    expect(screen.getByRole("button")).toHaveClass('MuiSvgIcon-root');
+  });
+
```
[the remark this answers is not in the document]
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js]: done

File: app/javascript/components/action-button/component.spec.js [hunk 5]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
...
+  });
 
   // Check if outlined class is set
-  it.todo("renders outlined button");
+  //it.todo("renders outlined button");
+
+  //it.todo("renders a pending state");
 
-  it.todo("renders a pending state");
+  //it.todo("triggers a passed cancel function");
```
[the remark this answers is not in the document]
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js]: done

File: app/javascript/components/action-button/components/default-button/component.spec.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { mountedComponent, screen } from "test-utils";
```
Comment (jtoliver-quoin) [inline app/javascript/components/action-button/components/default-button/component.spec.js:1]: Copy and paste these new files into the old files and rename to component.spec.js. This should keep the git history of the file.

File: app/javascript/components/action-button/components/icon-button/component.spec.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { mountedComponent, screen } from "test-utils";
+import IconButton from "./component";
```
Comment (jtoliver-quoin) [inline app/javascript/components/action-button/components/icon-button/component.spec.js:2]: Same here

File: app/javascript/components/action-button/components/link/component.spec.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { mountedComponent, screen } from "test-utils";
```
Comment (jtoliver-quoin) [inline app/javascript/components/action-button/components/link/component.spec.js:1]: Same here

File: app/javascript/components/action-button/components/default-button/component.spec.js (hunk 6 above)
Comment (gitdivyanshu) [reply app/javascript/components/action-button/components/default-button/component.spec.js:1]: done

File: app/javascript/components/action-button/components/icon-button/component.spec.js (hunk 7 above)
Comment (gitdivyanshu) [reply app/javascript/components/action-button/components/icon-button/component.spec.js:2]: done

File: app/javascript/components/action-button/components/link/component.spec.js (hunk 8 above)
Comment (gitdivyanshu) [reply app/javascript/components/action-button/components/link/component.spec.js:1]: done

File: app/javascript/components/action-button/component.spec.js [hunk 9]
```diff
@@ -35,15 +34,44 @@ describe("<ActionButton />", () => {
   });
 
   // Check if transparent class is set
-  it.todo("renders transparent button");
+  //it.todo("renders transparent button");
```
[the remark this answers is not in the document]
Comment (gitdivyanshu) [reply app/javascript/components/action-button/component.spec.js]: done

**Pull request merged**